torchgeo/datasets/loveda.py

```diff
@@ -0,0 +1,302 @@
+            os.path.join(self.directory, s.capitalize()) for s in self.scene
+        ]
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found at root directory or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.files = self._load_files(self.scene_paths, self.split)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            sample: image and mask at that index with image of dimension 3x1024x1024
+                    and mask of dimension 1024x1024
+
+        """
+        files = self.files[index]
+        image = self._load_image(files["image"])
+
+        if self.split != "test":
+            mask = self._load_target(files["mask"])
+            sample = {"image": image, "mask": mask}
+        else:
+            sample = {"image": image}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of datapoints in the dataset.
+
+        Returns:
+            length of dataset
+
+        """
+        return len(self.files)
+
+    def _load_files(self, scene_paths: List[str], split: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            scene_paths: contains one or two paths, depending on whether user has
+                         specified only 'rural', 'only 'urban' or both
+            split: subset of dataset, one of [train, val, test]
+
+        """
+        images = []
+
+        for s in scene_paths:
+            images.extend(glob.glob(os.path.join(s, "images_png", "*.png")))
+
+        images = sorted(images)
+
+        if self.split != "test":
+            masks = [image.replace("images_png", "masks_png") for image in images]
+            files = [
+                dict(image=image, mask=mask) for image, mask, in zip(images, masks)
+            ]
+        else:
+            files = [dict(image=image) for image in images]
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            tensor: the loaded image
+
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load a single mask corresponding to image.
+
+        Args:
+            path: path to the mask
```

Comment:
These should probably have an empty line between them

Reply:
is this necessary if pydocstyle isn't complaining?

Reply:
For some reason I recall pydocstyle complaining about this previously. At least this has happened to me when copy/pasting docstrings from other datasets sometimes the newlines will get stripped.

Reply:
You can easily build/view docs locally to check (e.g. I have a `make docs` in addition to `make tests` in my Makefile).

Reply:
The docs themselves also have a Makefile, it's just a lot of work to add a new remote, fetch, checkout this branch, build the docs, and load them when Sphinx has a single button feature that lets you view them directly.